src/test/java/lambda/part2/exercise/Exercise2.java

```diff
@@ -0,0 +1,100 @@
+package lambda.part2.exercise;
+
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.function.Predicate;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstName() {
+        //предикат Person -> boolean, проверяющий что имя и фамилия человека не пусты
+        Predicate<Person> validate = person -> !person.getLastName().isEmpty() && !person.getFirstName().isEmpty();
+
+        assertThat(validate.test(new Person("Алексей", "Доренко", 40)), is(true));
+        assertThat(validate.test(new Person("Николай", "", 30)), is(false));
+        assertThat(validate.test(new Person("", "Мельников", 20)), is(false));
+    }
+
+    // метод (Person -> boolean) -> (Person -> boolean)
+    // - возвращает новый предикат, являющийся отрицанием исходного
+    // - при реализации использовать логический оператор !
+    private Predicate<Person> negateUsingLogicalOperator(Predicate<Person> predicate) {
+        return (person) -> !predicate.test(person);
+    }
+
+    //  метод (Person -> boolean, Person -> boolean) -> (Person -> boolean)
+    //  - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
+    //  - при реализации использовать логический оператор &&
+    private Predicate<Person> andUsingLogicalOperator(Predicate<Person> left, Predicate<Person> right) {
+        return person -> left.test(person) && right.test(person);
+    }
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
+        Predicate<Person> personHasEmptyFirstName = person -> person.getFirstName().isEmpty() || person.getFirstName() == null;
+        Predicate<Person> personHasEmptyLastName = person -> person.getLastName().isEmpty() || person.getFirstName() == null;
+
+        Predicate<Person> personHasNotEmptyFirstName = negate(personHasEmptyFirstName);
```

Comment:
You need to use `negateUsingLogicalOperator(...)` here

Reply:
Done